### PR TITLE
Integration of recent launcher changes from ethan's branch

### PR DESCRIPTION
**Pull request by dbadb**
This pull request represents integration of ethan's slightly-dusty branch with the mainline.
It also includes the switchover to ButtonFactory.

I encourage all reviewers to focus on the areas that they individually own.

### CODE REVIEW

File: src/main/java/com/spartronics4915/frc2020/commands/IndexerCommands.java [hunk 1]
```diff
@@ -376,9 +385,5 @@ public LoadToLauncher(int ballsToShoot)
             );
         }
 
-        public LoadToLauncher()
-        {
-            this(1);
```
Comment (pietroglyph) [inline src/main/java/com/spartronics4915/frc2020/commands/IndexerCommands.java:381]: What’s the point of this change?

**Review by pietroglyph — APPROVED** [review]
Looks good! Thanks for doing the button switcher.

File: src/main/java/com/spartronics4915/frc2020/commands/IndexerCommands.java (hunk 1 above)
Comment (dbadb) [reply src/main/java/com/spartronics4915/frc2020/commands/IndexerCommands.java:381]: Someone was calling the constructor with no argument.  Not clear what the intent was, but elsewhere someone was doing something related to this, thus my inference.

**Comment by dbadb** [comment]
@pietroglyph @j-james - running gradle test on this result produced the following warnings:

    WARNING ButtonFactory button collision 0 3
    WARNING ButtonFactory button collision 0 1

which means that there are two commands that want joystick button 3 and 1,  I have no opinion about who should really get them.

**Comment by pietroglyph** [comment]
I went ahead and fixed this collision.

**Pull request merged**